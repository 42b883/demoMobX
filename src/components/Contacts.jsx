import React, { Component } from 'react';
import ContactItem from './ContactItem.jsx';


class Contacts extends Component {
    render() {
        let filteredContacts = this.props.contacts.filter(contact => {
                return contact.name.toLowerCase().includes(this.props.search.toLowerCase());
            }
        )
        let contactList = this.props.contacts.length ? (
            filteredContacts.map(contact => {
                return (
                    <ContactItem 
                    addToFav={this.props.addToFav}
                    key={contact.id}
                    id={contact.id}
                    contact={contact}
                    contacts={this.props.contacts}
                    name={contact.name}
                    email={contact.email}
                    username={contact.username}
                    website={contact.website}
                    phone={contact.phone}
                    hometown={contact.address.city}
                    />
                )
            })        

        ) : (
            <div className="center">Contacts list is empty</div>
        )


        //
        let favContactList = this.props.fav.length ? (
            filteredContacts.map(contact => {
                return (
                    <ContactItem 
                    addToFav={this.props.addToFav}
                    key={contact.id}
                    id={contact.id}
                    contact={contact}
                    contacts={this.props.contacts}
                    name={contact.name}
                    email={contact.email}
                    username={contact.username}
                    website={contact.website}
                    phone={contact.phone}
                    hometown={contact.address.city}
                    />
                )
            })        

        ) : (
            <div className="center">Favorite contacts list is empty</div>
        )

        return (
            <div>{contactList}</div>
        );
    }
}

export default Contacts
