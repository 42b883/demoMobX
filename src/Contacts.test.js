/* eslint-env test */

import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "mobx-react";
import { observable, action, decorate } from "mobx";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Contacts from "./components/Contacts.jsx";

configure({ adapter: new Adapter() });

it("renders without crashing", () => {
  // const contactsStore = new DecoratedContactsStore();
  const props = {
    ContactsStore: {
      contacts: [
        { id: 1, name: "Kelsey", username: "@kellly" },
        { id: 2, name: "John", username: "@danJ" }
      ]
    }
  };
  const wrapper = shallow(<Contacts {...props} />);

  expect(wrapper).toMatchSnapshot();
});

it("without component", () => {
  const wrapper = shallow(
    <Provider>
      <Contacts />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
