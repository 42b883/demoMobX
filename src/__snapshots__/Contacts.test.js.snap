// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Contacts
  ContactsStore={
    Object {
      "contacts": Array [
        Object {
          "id": 1,
          "name": "Kelsey",
          "username": "@kellly",
        },
        Object {
          "id": 2,
          "name": "John",
          "username": "@danJ",
        },
      ],
    }
  }
/>
`;

exports[`without component 1`] = `<inject-Contacts-with-ContactsStore />`;
